lib/Sema/SemaAPINotes.cpp

```diff
@@ -149,8 +171,14 @@ namespace {
       auto end = D->specific_attr_end<A>();
       auto existing = getExistingAttr(D);
       if (existing != end) {
-        // Remove the existing attribute.
+        // Remove the existing attribute, and treat it as a superseded
+        // non-versioned attribute.
+        auto *versioned =
+            SwiftVersionedAttr::CreateImplicit(S.Context, clang::VersionTuple(),
+                                               *existing);
+
         D->getAttrs().erase(existing.getCurrent());
+        D->addAttr(versioned);
```

Comment:
@DougGregor: This part is new since last time. Do you see any issues with this version?